File: solutions/javascript/etl/1/etl.js
export const transform = (old) => {
  const result = {};
  for (const [score, letters] of Object.entries(old)) {
    const points = parseInt(score, 10);
    for (const letter of letters) {
      result[letter.toLowerCase()] = points;
    }
  }
  return result;
};

File: solutions/python/etl/1/etl.py
"""Transform legacy Scrabble score data into the new one-to-one format."""

from typing import Dict, List


def transform(legacy_data: Dict[int, List[str]]) -> Dict[str, int]:
    """Convert a legacy score-to-letters mapping into a letter-to-score mapping.

    Args:
        legacy_data: A dictionary mapping point values to lists of uppercase letters.

    Returns:
        A dictionary mapping each lowercase letter to its point value.
    """
    new_format: Dict[str, int] = {}

    for score, letters in legacy_data.items():
        for letter in letters:
            new_format[letter.lower()] = score

    return new_format